### PR TITLE
Added spring extension

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: opentracing-spring-extension/opentracing-spring-web-extension/src/test/java/org/zalando/opentracing/spring/web/extension/ExampleController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @GetMapping("/async")
+    public CompletableFuture<String> async() {
+        return CompletableFuture.completedFuture("Later");
+    }
+
+    @GetMapping("/error")
+    @ResponseStatus(SERVICE_UNAVAILABLE)
+    public void error() {
```
Comment (whiskeysierra) [inline opentracing-spring-extension/opentracing-spring-web-extension/src/test/java/org/zalando/opentracing/spring/web/extension/ExampleController.java]: Codacy Issue found: [Document empty method body](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625671)

File: opentracing-spring-extension/opentracing-spring-web-extension-autoconfigure/src/test/java/org/zalando/opentracing/spring/web/extension/autoconfigure/OpenTracingWebExtensionAutoConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.DEFINED_PORT;
+import static org.springframework.http.HttpStatus.OK;
+
+@ExtendWith(SpringExtension.class) // only needed for Spring 4 support
+@SpringBootTest(
+        webEnvironment = DEFINED_PORT,
```
Comment (lukasniemeier-zalando) [inline opentracing-spring-extension/opentracing-spring-web-extension-autoconfigure/src/test/java/org/zalando/opentracing/spring/web/extension/autoconfigure/OpenTracingWebExtensionAutoConfigurationTest.java]: `RANDOM_PORT` and inject port number? I had troubles with tests on `8080` in the past

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: opentracing-spring-extension/opentracing-spring-web-extension-autoconfigure/src/test/java/org/zalando/opentracing/spring/web/extension/autoconfigure/OpenTracingWebExtensionAutoConfigurationTest.java (hunk 2 above)
Comment (whiskeysierra) [reply opentracing-spring-extension/opentracing-spring-web-extension-autoconfigure/src/test/java/org/zalando/opentracing/spring/web/extension/autoconfigure/OpenTracingWebExtensionAutoConfigurationTest.java]: > I had troubles with tests on `8080` in the past

Like what?

> `RANDOM_PORT` and inject port number?

I'd delay that until I see an issue. Injecting the port number adds even more boilerplate to the test.

**Pull request merged**